impl/src/main/java/org/jboss/weld/bean/proxy/InterceptedProxyFactory.java

```diff
@@ -142,35 +133,33 @@ protected void addMethodsFromClass(ClassFile proxyClassType, ClassMethod staticC
                     }
                 }
                 processedBridgeMethods.addAll(declaredBridgeMethods);
-                cls = cls.getSuperclass();
-            }
-            // We want to iterate over pre-defined interfaces (getAdditionalInterfaces()) and also over those we discovered earlier (interfacesToInspect)
-            Set<Class<?>> allInterfaces = new HashSet<>(getAdditionalInterfaces());
-            if (interfacesToInspect != null) {
-                allInterfaces.addAll(interfacesToInspect);
             }
-            for (Class<?> c : allInterfaces) {
-                for (Method method : c.getMethods()) {
-                    MethodSignature signature = new MethodSignatureImpl(method);
-                    // For interfaces we do not consider return types when going through processed bridge methods
-                    try {
-                        if (isMethodAccepted(method, getProxySuperclass()) && enhancedMethodSignatures.contains(signature)
-                            && !finalMethods.contains(signature) && !processedBridgeMethods.contains(signature)) {
-
-                            final MethodInformation methodInfo = new RuntimeMethodInformation(method);
-                            ClassMethod classMethod = proxyClassType.addMethod(method);
-                            if (interceptedMethodSignatures.contains(signature) && Reflections.isDefault(method)) {
-                                // this method is intercepted and is default
-                                createInterceptedMethod(classMethod, methodInfo, method, staticConstructor);
-                                BeanLogger.LOG.addingMethodToProxy(method);
-                            } else {
-                                createNotInterceptedMethod(classMethod, methodInfo, method, staticConstructor);
+            if (builtFromInterface) {
+                // since we are just on top of an interface, we will need to add all the methods from interface
+                // hierarchy so that there are no AbstractMethodError exception popping up
+                for (Class<?> c : interfacesToInspect) {
+                    for (Method method : c.getMethods()) {
+                        MethodSignature signature = new MethodSignatureImpl(method);
+                        try {
+                            if (isMethodAccepted(method, getProxySuperclass())
```

Comment:
I am not really sure what is the correct check here. I have simplified it a lot because we are operating on method which belong to interface classes only. Compared to previous solution:

* Using `enhancedMethodSignatures.contains(signature)` won't work as those methods are not part of the signatures we know. The reason is that signatures seem to be built only from impl methods and default methods in interfaces.
* Removed `!finalMethods.contains(signature)` is pointless, there cannot be final methods on interfaces
* Retained `!processedBridgeMethods.contains(signature)` - not sure this was a smart move, TBF I am not 100% we can see bridge method on an interface?

Reply:
Hm, even interfaces can have bridge methods. So it's ok to retain the check.